framework/src/utils/InputParameters.C

```diff
@@ -1195,3 +1230,61 @@ InputParameters::varName(const std::string & var_param_name,
 
   return variable_name;
 }
+
+void
+InputParameters::renameParam(const std::string & old_name,
+                             const std::string & new_name,
+                             const std::string & new_docstring)
+{
```

Comment:
I guess if we want to remove the old name completely for a parameter in the future, we can just edit the name in `addParam` functions and the docstring and remove this function?